ingestion-core/src/main/java/com/mozilla/telemetry/ingestion/core/transform/PubsubMessageToObjectNode.java

```diff
@@ -39,14 +38,16 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ExecutionException;
+import java.util.function.BiFunction;
 import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 /**
- * Transform a {@link PubsubMessage} into an {@link ObjectNode}.
+ * Transform attributes and data into an {@link ObjectNode}.
  */
-public abstract class PubsubMessageToObjectNode implements Function<PubsubMessage, ObjectNode> {
+public abstract class PubsubMessageToObjectNode
+    implements BiFunction<Map<String, String>, byte[], ObjectNode> {
```

Comment:
To avoid choosing between beam and google cloud java sdk implementations of `PubsubMessage`, I modified this to accept attributes and data, instead of message.